protocols/isis/server/lsdb.go

```diff
@@ -312,3 +312,39 @@ func (l *lsdb) sendCSNPs(ifa *netIfa) {
 		ifa.sendCSNP(&c, l.level())
 	}
 }
+
+func (l *lsdb) processLSP(ifa *netIfa, lspdu *packet.LSPDU) {
+	log.Debug("Processing received LSP")
+	l.lspsMu.Lock()
+	defer l.lspsMu.Unlock()
+
+	existingLSDBEntry, exists := l.lsps[lspdu.LSPID]
+	if !exists || lspdu.SequenceNumber > existingLSDBEntry.lspdu.SequenceNumber {
+		log.Debugf("ISIS: Received newer LSPDU %v sequence number %d", lspdu.LSPID, lspdu.SequenceNumber)
+		l.processNewerLSPDU(ifa, lspdu)
+		return
+	}
+
+	if lspdu.SequenceNumber == existingLSDBEntry.lspdu.SequenceNumber {
```

Comment:
This could use an `exists` guard to be safe, and potentially an "I cannot let you do this Dave" case for an invalid LSPDU

Reply:
This is safe guarded. If !exists it will exit above.